app/src/main/kotlin/io/homeassistant/companion/android/launch/my/MyLinkHandler.kt

```diff
@@ -0,0 +1,131 @@
+            FailFast.fail {
+                "Invalid deep link double check the host or the scheme: $uri"
+            }
+            return LinkDestination.NoDestination
+        }
+
+        return when {
+            path.startsWith(REDIRECT_URL_PATH) -> handleNavigateDeepLink(uri)
+            path.startsWith(INVITE_URL_PATH) -> handleInviteUniversalLink(uri)
+            else -> {
+                FailFast.fail { "Unknown or invalid deep link: $uri" }
+                LinkDestination.NoDestination
+            }
+        }
+    }
+
+    /**
+     * Attempts to extract the target Home Assistant instance URL from the fragment part of the provided URI.
+     *
+     * The expected invitation link format is:
+     * `https://my.home-assistant.io/invite#url=http://homeassistant.local:8123`
+     *
+     * The target URL is embedded in the fragment for security reasons,
+     * preventing it from being sent to `my.home-assistant.io`. This function extracts the
+     * `url` parameter from the fragment.
+     *
+     * @param uri The URI to process containing the invitation link.
+     * @return [LinkDestination.Onboarding] if a valid target URL is successfully extracted from the `uri`'s fragment,
+     *         otherwise [LinkDestination.NoDestination].
+     */
+    private fun handleInviteUniversalLink(uri: Uri): LinkDestination {
+        // Extract the server URL from the fragment #url=http://homeassistant.local:8123 and make it queryParam in a fake URI
+        val serverURL = "https://?${uri.encodedFragment.orEmpty()}".toUri().takeIf {
+            // If for some reason the fragment contains an opaque URI getQueryParameter will throw
+            !it.isOpaque
+        }?.getQueryParameter("url")
+
+        return if (serverURL == null) {
+            FailFast.fail { "Deep link does not contains a valid URL to a server ($uri)" }
+            LinkDestination.NoDestination
+        } else {
+            LinkDestination.Onboarding(serverURL)
+        }
+    }
+
+    private fun handleNavigateDeepLink(uri: Uri): LinkDestination {
+        if (!serverManager.isRegistered()) {
+            Timber.w("No server registered, cannot handle deep link")
+            return LinkDestination.NoDestination
+        }
+
+        if (uri.getQueryParameter(MOBILE_PARAM) == MOBILE_VALUE) {
+            FailFast.fail { "$MOBILE_PARAM flag already present and set to 1, the deep link is already handled" }
+            return LinkDestination.NoDestination
+        }
+
+        val path = uri.buildUpon()
+            // We strip last / to handle old links created before https://github.com/home-assistant/frontend/pull/25841
+            // A trailing slash is always added by Netlify that is used to host https://my.home-assistant.io/, but
+            // the frontend was not supporting having a trailing slash before https://github.com/home-assistant/frontend/pull/25841
+            // in order to backward compatible we remove the trailing slash here.
```

Comment:
This is more than one sentence 


```suggestion
            // We strip the last / to handle old links created before https://github.com/home-assistant/frontend/pull/25841.
            // A trailing slash is always added by Netlify that is used to host https://my.home-assistant.io/, but
            // the frontend did not support having a trailing slash before https://github.com/home-assistant/frontend/pull/25841.
            // For backward compatibility, we remove the trailing slash here.
```